### PR TITLE
Consensus Test: Add testing version v39test

### PR DESCRIPTION
**Pull request by gmalouf**
## Summary

Introduce testing version v39test as we exercise changes targeted for forthcoming upgrade. Includes tweaks to tests where warranted for them to handle version changes/no longer depend on vFuture.

## Test Plan

Existing tests should pass.

### CODE REVIEW

**Review by algorandskiy — APPROVED** [review]

File: agreement/player_permutation_test.go [hunk 1]
```diff
@@ -811,16 +811,16 @@ func TestPlayerPermutation(t *testing.T) {
 }
 
 func playerPermutationCheck(t *testing.T, enableDynamicFilterTimeout bool) {
-	// create a protocol version where dynamic filter is enabled
-	version, _, configCleanup := createDynamicFilterConfig()
+	// create a protocol disabledFilterTimeoutVersion where dynamic filter is disabled
+	disabledFilterTimeoutVersion, _, configCleanup := overrideConfigWithDynamicFilterParam(false)
```
Comment (jannotti) [inline agreement/player_permutation_test.go:815]: I'm confused why `false` is hardcoded.  Isn't the idea to be able to check both ways, based on the argument?  I think I would give `enabledFilterTimeoutVersion` to this call, and then remove the check on line 822. Instead, always set the `inMsg.Proto`.

File: agreement/player_test.go [hunk 2]
```diff
@@ -37,7 +37,7 @@ func init() {
 }
 
 func makeTimeoutEvent() timeoutEvent {
-	return timeoutEvent{T: timeout, RandomEntropy: crypto.RandUint64()}
+	return timeoutEvent{T: timeout, RandomEntropy: crypto.RandUint64(), Proto: ConsensusVersionView{Version: protocol.ConsensusCurrentVersion}}
```
Comment (jannotti) [inline agreement/player_test.go:40]: It seem like this change means all tests with the dynamic filter turned on, where they used to run with it turned off. (Actually, I don't know how `Proto` was set before.)

File: data/transactions/logic/assembler_test.go [hunk 3]
```diff
@@ -536,7 +541,7 @@ func TestAssemble(t *testing.T) {
 	}
 }
 
-var experiments = []uint64{pairingVersion, spliceVersion}
+var experiments []uint64
```
Comment (jannotti) [inline data/transactions/logic/assembler_test.go:544]: Minor quibble: I'd prefer it keeps the same form, even while empty.  People are more likely to update it.
```suggestion
var experiments []uint64 = []uint64{}
```
unless some linter complains.

File: data/transactions/logic/eval_test.go [hunk 4]
```diff
@@ -1255,6 +1259,7 @@ func TestGlobal(t *testing.T) {
 		8:  {CallerApplicationAddress, globalV8TestProgram},
 		9:  {CallerApplicationAddress, globalV9TestProgram},
 		10: {AssetOptInMinBalance, globalV10TestProgram},
+		11: {AssetOptInMinBalance, globalV11TestProgram},
```
Comment (jannotti) [inline data/transactions/logic/eval_test.go:1262]: I feel like I'm seeing a bug that's my fault. We did add another global, `GenesisHash` so I think that should be here.  I'll look more closely in master.

File: data/transactions/logic/langspec_v1.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "Version": 1,
-  "LogicSigVersion": 9,
+  "LogicSigVersion": 10,
```
Comment (jannotti) [inline data/transactions/logic/langspec_v1.json:3]: Looking at this now, I suppose the fix is to remove `LogicSigVersion` entirely. I don't see the value with `Version` right there doing the same thing.  But I think we said we'd do that in master.

File: ledger/testing/consensusRange.go [hunk 6]
```diff
@@ -59,7 +59,8 @@ var consensusByNumber = []protocol.ConsensusVersion{
 	protocol.ConsensusV35, // minor, double upgrade withe v34
 	protocol.ConsensusV36, // AVM v8, box storage
 	protocol.ConsensusV37,
-	protocol.ConsensusV38, // AVM v9, ECDSA pre-check, stateproofs recoverability
+	protocol.ConsensusV38,     // AVM v9, ECDSA pre-check, stateproofs recoverability
+	protocol.ConsensusV39test, // AVM v10, loogicsig opcode budget pooling, elliptic curve ops, dynamic round times
```
Comment (jannotti) [inline ledger/testing/consensusRange.go:63]:
```suggestion
	protocol.ConsensusV39test, // AVM v10, logicsig opcode budget pooling, elliptic curve ops, dynamic round times
```

File: data/transactions/logic/langspec_v1.json (hunk 5 above)
Comment (gmalouf) [reply data/transactions/logic/langspec_v1.json:3]: I think we landed on doing it in master, agreed.

File: data/transactions/logic/eval_test.go (hunk 4 above)
Comment (jannotti) [reply data/transactions/logic/eval_test.go:1262]: I made a PR to fix this in master. It's just a lack of a test, so there's not reason to make any changes here.

**Review by jannotti — APPROVED** [review]

**Pull request merged**